include/xsimd/types/xsimd_traits.hpp

```diff
@@ -11,6 +11,7 @@
 
 #include <type_traits>
 
+#include "../types/xsimd_base.hpp"
```

Comment:
This file is not supposed to be included directly, `xsimd_types_include.hpp` is responsible for including the required batches and related functions. So either the types / functions defined in `xsimd_base.hpp` should be moved to `xsimd_utils.hpp` or the files defining math functions for batches should only be included when simd instructions are enabled. I find this last option cleaner, however it is a bigger (potentially breaking) change.

Reply:
as discussed, this is for the next big refactoring